### DISCUSSION TITLE
DynamoDB Database Support

### DISCUSSION THREAD

**Discussion by farzadpanahi** [grafana/grafana#41214] [Platform]
As of today [supported databases](https://grafana.com/docs/grafana/latest/installation/requirements/#supported-databases) are:

- SQLite
- MySQL
- PostgreSQL

In an attempt to make highly available Grafana cheaper to run on AWS, I think it would be really helpful to support serverless databases with on-demand capacity set-up, like DynamoDB to store Grafana configuration. 

Right now the minimum cost to run a minimal RDS Multi-AZ database in AWS us-east is $28.39:

> $0.032 per RDS db.t4g.micro Multi-AZ instance hour (or partial hour) running PostgreSQL 743.497 Hrs   $23.79
> $0.230 per GB-month of provisioned GP2 storage for Multi-AZ deployments 19.986 GB-Mo   $4.60

Since Grafana's database size is small and there is no need for many read/write queries, it would be way cheaper to keep the config on an [on-demand capacity DynamoDB](https://aws.amazon.com/dynamodb/pricing/on-demand/). The cost should be just cents or a few dollars for low volume.

For instance it would only cost $0.25/month to run the following on DynamoDB:

>  - Storage size: 1GB 
>  - Avg. item size: 1KB 
>  - Standard on-demand writes per hour: 5
>  - Eventually consistent on-demand reads per hour: 10   

I think supporting DynamoDB helps users to easily start with a proper highly-available and scalable setup for Grafana on AWS with minimum cost.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
